### PR TITLE
Fix E2E Happy path tests to pass debug verification step

### PR DESCRIPTION
**Pull request by dmytro-ndp**
#### 1. Fixes E2E Happy path tests to pass step "_Run debug and check application stop in the breakpoint_" (#18034)

Workaround log:
```
...
12:04:42           ▼ Waiting for debugger to connect (threads to appear in "Threads" view)
12:04:42           ▼ Workaround for the https://github.com/eclipse/che/issues/18034 issue.
12:04:42           ▼ Click on "Threads" view title
12:04:42           ▼ Waiting for debugger to connect (threads to appear in "Threads" view)
12:04:42           ▼ Editor.selectTab "WelcomeController.java"
...
```

#### 2. Changes E2E Happy path workspace devfile to use _quay.io_ images instead of ones on _Docker Hub_ (#18426)
- quay.io/eclipse/che--centos--postgresql-96-centos7:9.6-b681d78125361519180a6ac05242c296f8906c11eab7e207b5ca9a89b6344392
- quay.io/mloriedo/spring-petclinic

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/18034, https://github.com/eclipse/che/issues/18426

### How to test this PR?
Run Happy path tests.

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [x] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [x] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Comment by dmytro-ndp** [comment]
[crw-ci-test]

File: tests/e2e/pageobjects/ide/DebugView.ts [hunk 1]
```diff
@@ -52,12 +52,25 @@ export class DebugView {
     async waitForDebuggerToConnect(timeout: number = TimeoutConstants.TS_DEBUGGER_CONNECTION_TIMEOUT) {
         await this.driverHelper.getDriver().wait(async () => {
             Logger.debug(`Waiting for debugger to connect (threads to appear in "Threads" view)`);
-            const threadElements: WebElement[] = await this.driverHelper.getDriver().findElements(By.xpath(`//div[contains(@class, 'theia-debug-thread')]`));
+
+            const threadsTreeLocator = `//div[contains(@class, 'theia-debug-thread')]`;
+
+            const threadElements: WebElement[] = await this.driverHelper.getDriver().findElements(By.xpath(threadsTreeLocator));
```
Comment (Ohrimenko1988) [inline tests/e2e/pageobjects/ide/DebugView.ts]: maybe simplified by ```driverHelper.waitAllPresence```

**Review by Ohrimenko1988 — APPROVED** [review]

**Review by rhopp — APPROVED** [review]

File: tests/e2e/pageobjects/ide/DebugView.ts (hunk 1 above)
Comment (dmytro-ndp) [reply tests/e2e/pageobjects/ide/DebugView.ts]: thanks for notice!
It has been applied.

Comment (dmytro-ndp) [reply tests/e2e/pageobjects/ide/DebugView.ts]: thanks for notice!
It has been applied.

**Comment by benoitf** [comment]
Hi, does it mean that it's waiting 30seconds before seeing that it's collapsed and then clicking on the entry to make it expanded 🤔  ?

**Pull request merged**